go/tosca/types_test.go

```diff
@@ -87,15 +87,61 @@ func TestAddress_JSON_InvalidValueDecodingFails(t *testing.T) {
 	}
 }
 
+func TestValue_NewValue(t *testing.T) {
+
+	tests := []struct {
+		value Value
+		index int
+	}{
+		{NewValue(1), 31},
+		{NewValue(1, 0), 23},
+		{NewValue(1, 0, 0), 15},
+		{NewValue(1, 0, 0, 0), 7},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%v[%d]", test.value, test.index), func(t *testing.T) {
+
+			if test.value[test.index] != 1 {
+				t.Errorf("NewValue failed to set the correct value.")
+			}
+
+		})
+	}
+}
+
+func TestValue_Get(t *testing.T) {
+
+	tests := []struct {
+		value Value
+		index int
+		want  uint64
+	}{
+		{NewValue(1), 0, 1},
+		{NewValue(1, 0), 1, 1},
+		{NewValue(1, 0, 0), 2, 1},
+		{NewValue(1, 0, 0, 0), 3, 1},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%v[%d]", test.value, test.index), func(t *testing.T) {
+			if got := test.value.getInternalUint64(test.index); test.want != got {
+				t.Errorf("wanted %v but got %v.", test.want, got)
+			}
+		})
+	}
+}
```

Comment:
The two new tests are checking that the provides `uint64` values are converted to the 32 bytes in the type using big endian order, but I think there should be a test that also makes sure that for the arithmetic operations this order is used. Could be something simpler like
```
Sub(NewValue(1,0),NewValue(1)) == NewValue(math.MaxUint64)
Sub(NewValue(1,0,0),NewValue(1)) == NewValue(math.MaxUint64,math.MaxUint64)
Sub(NewValue(1,0,0,0),NewValue(1)) == NewValue(math.MaxUint64,math.MaxUint64,math.MaxUint64)
```
and the same for `Add`.

Reply:
tests added. nice suggestion